### PR TITLE
[JIMT][CT-835] - added keyboard traps to dialogs

### PR DESCRIPTION
**Pull request by thomasoniii**
Our dialogs from lab2's dialogControl were keyboard accessible with tabs, but wouldn't automatically respond to `Enter` (for the command button) or `Escape` (for the cancel button).

Extends the dialogs with a simple keyboard trap to respond to those keypresses when the dialogs are open.

## Links

- jira ticket: [CT-835](https://codedotorg.atlassian.net/browse/CT-835)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lab2/views/dialogs/GenericPrompt.tsx [hunk 1]
```diff
@@ -91,8 +91,8 @@ const GenericPrompt: React.FunctionComponent<GenericPromptProps> = ({
         confirm: {
           callback: () => handleConfirm?.(prompt),
           disabled:
-            requiresPrompt !== false &&
-            (Boolean(errorMessage) || !prompt.length),
+            Boolean(errorMessage) ||
+            (requiresPrompt === true && !prompt.length),
```
Comment (thomasoniii) [inline apps/src/lab2/views/dialogs/GenericPrompt.tsx]: This is unrelated to the rest of the change and was an opportunistic bug fix - the logic was wrong.

Previously, if the prompt.length was zero _and_ a prompt was not required, it'd be enabled. However, if there's an error condition, then it always needs to be disabled.

Comment (molly-moen) [reply apps/src/lab2/views/dialogs/GenericPrompt.tsx]: Can we just write `requiresPrompt` rather than `requiresPrompt === true`?

**Review by molly-moen — APPROVED** [review]
